pkg/controllers/user/authz/podsecuritypolicy/binding.go

```diff
@@ -0,0 +1,140 @@
+package podsecuritypolicy
+
+import (
+	"fmt"
+
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/extensions/v1beta1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	v12 "github.com/rancher/types/apis/rbac.authorization.k8s.io/v1"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	v1beta13 "k8s.io/api/extensions/v1beta1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RegisterBindings updates the pod security policy for this binding if it has been changed.  Also resync service
+// accounts so they pick up the change.  If no policy exists then exits without doing anything.
+func RegisterBindings(context *config.UserContext) {
+	logrus.Infof("registering podsecuritypolicy project handler for cluster %v", context.ClusterName)
+
+	lifecycle := &lifecycle{
+		policyLister:         context.Extensions.PodSecurityPolicies("").Controller().Lister(),
+		policies:             context.Extensions.PodSecurityPolicies(""),
+		psptLister:           context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		clusterRoleLister:    context.RBAC.ClusterRoles("").Controller().Lister(),
+		clusterRoles:         context.RBAC.ClusterRoles(""),
+		serviceAccountLister: context.Core.ServiceAccounts("").Controller().Lister(),
+		serviceAccounts:      context.Core.ServiceAccounts("").Controller(),
+	}
+
+	context.Management.Management.PodSecurityPolicyTemplateProjectBindings("").
+		AddLifecycle("PodSecurityPolicyTemplateProjectBindingsLifecycleHandler", lifecycle)
+}
+
+type lifecycle struct {
+	policyLister         v1beta1.PodSecurityPolicyLister
+	policies             v1beta1.PodSecurityPolicyInterface
+	psptLister           v3.PodSecurityPolicyTemplateLister
+	clusterRoleLister    v12.ClusterRoleLister
+	clusterRoles         v12.ClusterRoleInterface
+	serviceAccountLister v1.ServiceAccountLister
+	serviceAccounts      v1.ServiceAccountController
+}
+
+func (l *lifecycle) Create(obj *v3.PodSecurityPolicyTemplateProjectBinding) (*v3.PodSecurityPolicyTemplateProjectBinding, error) {
+	return l.sync(obj)
+}
+
+func (l *lifecycle) Updated(obj *v3.PodSecurityPolicyTemplateProjectBinding) (*v3.PodSecurityPolicyTemplateProjectBinding, error) {
+	return l.sync(obj)
+}
+
+func (l *lifecycle) Remove(obj *v3.PodSecurityPolicyTemplateProjectBinding) (*v3.PodSecurityPolicyTemplateProjectBinding, error) {
+	return obj, resyncServiceAccounts(l.serviceAccountLister, l.serviceAccounts, "")
```

Comment:
Since a binding is for a particular project and a project is a subset of the cluster's namespaces, enqueueing all service accounts in the cluster is overly aggressive. You should get all the namespaces in the project and then just enqueue the svc accounts for those namespaces. (You can actually just enqueue those namespaces themselves and they will in turn enquque their service accounts when the namespace handler you wrote gets executed)